src/zocalo/service/dispatcher.py

```diff
@@ -0,0 +1,217 @@
+import copy
+import errno
+import os
+import timeit
+import uuid
+import pkg_resources
+
+from pprint import pformat
+
+import workflows.recipe
+from workflows.services.common_service import CommonService
+
+
+class Dispatcher(CommonService):
+    """
+    Single point of contact service that takes in job meta-information
+    (say, a data collection ID), a processing recipe, a list of recipes,
+    or pointers to recipes stored elsewhere, and mangles these into something
+    that can be processed by downstream services.
+    """
+
+    # Human readable service name
+    _service_name = "Dispatcher"
+
+    # Logger name
+    _logger_name = "services.dispatcher"
+
+    # Define a base path where your recipes are located (accessed with zocalo.go -r $recipename)
+    recipe_basepath = None
+
+    def filter_load_recipes_from_files(self, message, parameters):
```

Comment:
I guess you got Tim to pull this (https://github.com/DiamondLightSource/python-dlstbx/blob/master/src/dlstbx/services/dispatcher.py#L224, and apply_parameters https://github.com/DiamondLightSource/python-dlstbx/blob/master/src/dlstbx/services/dispatcher.py#L255) out of the dlstbx code into filters because they are executed in the same way as other filters, but for applying parameters contrary to what you have said before order does matter. The parameters must be applied _last_ otherwise you get unevaluated placeholders. For this you cannot rely on entry point order. I dont know how recipe loading is effected in this chain and how sensitive it is to order.